modules/ml-training/training-images/mnist/deployspec.yaml

```diff
@@ -0,0 +1,25 @@
+deploy:
+  phases:
+    build:
+      commands:
+      - export TIMESTAMP=$(date +%Y%m%d_%H%M%S)
+      - export IMAGE_TAG="build-$TIMESTAMP" && echo IMAGE_TAG=$IMAGE_TAG
+      - export ECR_REGISTRY_URL=$AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com
+      - export REPOSITORY_URI=$ECR_REGISTRY_URL/$ADDF_PARAMETER_ECR_REPOSITORY_NAME && echo "REPOSITORY_URI=$REPOSITORY_URI"
+      - export SAGEMAKER_BASE_IMAGE=763104351884.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/pytorch-training:2.1.0-gpu-py310-cu121-ubuntu20.04-ec2 && echo SAGEMAKER_BASE_IMAGE=$SAGEMAKER_BASE_IMAGE
+      # Log into SageMaker ECR Repo (https://github.com/aws/deep-learning-containers/blob/master/available_images.md)
+      - aws ecr get-login-password --region $AWS_DEFAULT_REGION | docker login --username AWS --password-stdin 763104351884.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com
+      - echo "docker build -t $REPOSITORY_URI:latest . --build-arg BASE_IMAGE=$SAGEMAKER_BASE_IMAGE"
+      - cd src && docker build -t $REPOSITORY_URI:latest . --build-arg BASE_IMAGE=$SAGEMAKER_BASE_IMAGE
+      # - Building the Docker image(s)...
+      - aws ecr get-login-password --region $AWS_DEFAULT_REGION | docker login --username AWS --password-stdin $ECR_REGISTRY_URL
+      - docker tag $REPOSITORY_URI:latest $REPOSITORY_URI:$IMAGE_TAG
+      - docker push $REPOSITORY_URI:latest && docker push $REPOSITORY_URI:$IMAGE_TAG
```

Comment:
I do not see where you are creating the ECR repo....and it is not passed in.  I am don't understand how tis module works with other modules as you are not creating nor exporting and ecr repo arn for reuse.  I believe this line will fail due to lack of a repo....